### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by stoneman**
Resolves: https://github.com/github/npm/issues/1346

Modify the docs to make them as complete/correct as is reasonably possible with the understanding I've gained from recent investigations into the code. 

It would be ideal to have this reviewed by:
 - someone who knows very little about npms-analyzer / our search
 - whoever knows the most about npms-analyser / our search

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -1,203 +1,261 @@
...
+- Extract package name, scope, version, description and keywords
+- Extract package author, publisher, maintainers and contributors
 - Extract the license
 - Get releases timing information
 - Extract repository and homepage
 - Extract README
 - Extract the package dependencies
 - Check if the package is deprecated
 - Check if the package has a test script
+- Check if the package contains "selective files" (whatever they are...?? https://github.com/npm/npms-analyzer/blob/master/lib/analyze/collect/metadata.js#L322)
```
Comment (radiantspace) [inline docs/architecture.md:65]: nit: maybe use concat into reference link?

[whatever they are...??](https://github.com/npm/npms-analyzer/blob/master/lib/analyze/collect/metadata.js#L322)

File: docs/architecture.md [hunk 2]
```diff
@@ -1,203 +1,261 @@
...
+- Fetch the repository activity in terms of commits
 - Check the build status
 
-This collector is susceptible to the GitHub [rate limit](https://developer.github.com/v3/rate_limit/) policy. To fight against this limit, you may define several GitHub keys in the config to be used in a round-robin fashion.
+This collector is susceptible to the GitHub [rate limit](https://developer.github.com/v3/rate_limit/) policy. To fight against this limit, several GitHub keys should be specified in the config. [token-dealer](https://github.com/moxystudio/node-token-dealer) is used to cycle through the keys and, when a key is rate limited, to remove it from the pool until it's ban will be lifted and to retry the request with a different key.
 
 ##### npm
 
-The npm collector uses the replicated CouchDB views and the npm [download-counts](https://github.com/npm/download-counts) API to extract useful information present there.
+The npm collector obtains data from [CouchDB](http://couchdb.apache.org/) views and the [download-counts](https://github.com/npm/download-counts) API.
```
Comment (radiantspace) [inline docs/architecture.md:92]: I think downloads count should hit the actual redis-download-counts now - https://github.com/npm/redis-download-counts, not the archived repo service.

**Review by radiantspace — APPROVED** [review]
Thanks for sorting things here and going through all the relevant documentation! Good job!

File: docs/architecture.md [hunk 3]
```diff
@@ -1,203 +1,261 @@
...
-- Fetch the number of issues and their distribution over time
 - Extract the homepage
+- Get number of stars, subscribers and forks
+- Fetch the number of issues/pull requests and their distribution over time
 - Fetch contributors
+- Fetch the repository activity in terms of commits
 - Check the build status
 
-This collector is susceptible to the GitHub [rate limit](https://developer.github.com/v3/rate_limit/) policy. To fight against this limit, you may define several GitHub keys in the config to be used in a round-robin fashion.
+This collector is susceptible to the GitHub [rate limit](https://developer.github.com/v3/rate_limit/) policy. To fight against this limit, several GitHub keys should be specified in the config. [token-dealer](https://github.com/moxystudio/node-token-dealer) is used to cycle through the keys and, when a key is rate limited, to remove it from the pool until it's ban will be lifted and to retry the request with a different key.
```
Comment (john7doe) [inline docs/architecture.md:88]: Have we reached out to someone on github to see if would be possible to have them assign a more lax policy the the now in-house npm team?

**Comment by john7doe** [comment]
Great job documenting this. We are in a much better position to work on this now :-)

**Review by john7doe — APPROVED** [review]

File: docs/architecture.md (hunk 3 above)
Comment (stoneman) [reply docs/architecture.md:88]: We have not yet but there is a feature to re-enable github integration which _may_ get worked on in the future and that will include looking at the rate limiting.

**Comment by stoneman** [comment]
Thanks for the reviews @radiantspace & @john7doe 🙂

File: docs/architecture.md (hunk 3 above)
Comment (radiantspace) [reply docs/architecture.md:88]: We're actually planning to do that for readmes rendering with GitHub public API, so there will be more clarity after we complete this story - https://github.com/github/npm/issues/1506

**Pull request merged**